src/ipc/ipc3/helper.c

```diff
@@ -278,11 +278,7 @@ static void comp_specific_builder(struct sof_ipc_comp *comp,
 	case SOF_COMP_NONE:
 		config->process.type = proc->type;
 		config->process.size = proc->size;
-#if CONFIG_LIBRARY || UNIT_TEST
-		config->process.data = proc->data + comp->ext_data_length;
-#else
-		config->process.data = proc->data;
-#endif
+		config->process.data = NULL;
```

Comment:
it is not only for NONE, it is for ten component cases, check line :268-278.
Are you sure all above cases all does not need this?